astropy/io/fits/scripts/fitscheck.py

```diff
@@ -173,15 +170,25 @@ def update(filename):
     Also updates fixes standards violations if possible and requested.
     """
 
-    hdulist = fits.open(filename, do_not_scale_image_data=True)
+    tmpfile = filename + '.bak'
```

Comment:
Would python's `tempfile` module be useful here?

Reply:
No strong preference for me (I did that quickly to debug the windows issue). One advantage of not using `/tmp` is that it can avoid a copy to another disk, e.g. if your data is on a NFS share.

Reply:
It's really bugging me that we need a temporary file at all. Is there no way to run a complete check without _actually_ writing it to a file? Or is the intention here to fix the problems?

Sorry I haven't had the chance to look into this more closely, but I'll do so next week.

Reply:
The goal here is to update the file with new checksums,  and windows does not allow to overwrite the file while it is open.

Reply:
Then why not read, close then write?

Reply:
Because there is no nice way to force reading the data (you need to iterate on all HDUs and access the `.data` attribute). The other (better) option would be to use the `update` mode and `.flush`, I think it can handle the checksums, but there may be a good reason why it is not used. So, at least for now, the tmp file and replace solution seems the most straightforward.